README.md
# Q_O
Voice helper
Исходный код, не рекомендую в нём копатся, комментариев нет, и код только я понимаю))) Если не боитесь лезть в дебри, милости прошу, моё дело предупредить
https://drive.google.com/file/d/12kxbVyK9Q6JSZbgBNSRrKvLMioicDkvU/view?usp=drive_link


history.txt
Start file
Ты: почему от молчать
Ты: экран
Ты: по
Ты: так ты без меня
Ты: я
Ты: а
15:01:44 Бот: б
Ты: привет
15:01:47 Бот: Да, привет.
Ты: кривит
Ты: а клыки
Ты: его
Ты: пива
Ты: ок
Ты: как и окуня я не пробил и
Ты: и между коки
Ты: привет давно не виделись
Ты: привет
20:40:25 Бот: Да, привет.
Ты: можешь тебя дописать
Ты: надо бы тебя чему-нибудь ещё и научить
Ты: пока брента работы свёрнутой много не работает накручивает часа
Ты: основном
Ты: что с тобой делать
20:41:14 Бот: Не бой, а боль!
Ты: я до сих пор остались фламенко и для того чтобы тебя учить
Ты: мне это нравится
Ты: вести ти энд заводи
Ты: оу появились к
Ты: надо найти бой
20:42:03 Бот: Не бой, а боль!
Ты: сука
Ты: вот я об этом и говорю
Ты: он тексте
Ты: детьми
Ты: а не проще ли просто коля как метод так и не надо
Ты: я методы радио которого племянниц нам ники
Ты: тебя лучше надо
Ты: переведи слова
Ты: нам с тобой
Ты: бой
20:47:32 Бот: Не бой, а боль!
Ты: принципе работает
Ты: по сделать метод херст мёртв
Ты: таймер и актёрскому
Ты: и
Ты: почти тоже самое только
Ты: ладно
Ты: копировать
Ты: новый яйца короче
Ты: не правда
Ты: ли
Ты: делом
Ты: ну
Ты: туры
Ты: хребту флоуз
Ты: перед запусти они авиньон
Ты: простите
20:49:11 Бот: Бог простит, хотя в мои алгоритмы не было записано кто такой бог
Ты: время
20:49:18 Бот: время 20:49:18, дата 2023-01-06 День недели Пятница
Ты: по-моему спрашивал бака время ладно
20:49:31 Бот: В штанах прохладно
Ты: убили негра
20:49:36 Бот: Ай ай ай ай ай я я я я я а убили негра, убили негра убили, ай ай ай я не за что не прочто, суки замочили
Ты: и делать этот привет есть
Ты: три пока
Ты: и
Ты: лар лер с юрт
Ты: переведи лет
20:50:16 Бот: years old
Ты: что
Ты: переведи цель
20:50:26 Бот: goal
Ты: переведи цель миссии
20:50:32 Бот: The purpose of the mission
Ты: ха
20:50:36 Бот: Хахахахахахаххах
Ты: переведи целевая слова
20:50:43 Бот: targeted words
Ты: уота таргетинга таргет алтарь таргитай было анхель
Ты: я запусти
Ты: перезапустить
Ты: привет давно не привет давно не виделись
20:51:38 Бот: Да, привет.
Ты: ага
Ты: кривя
Ты: а это если отношения чуть ниже вещах
Ты: какое расширение экрана
Ты: расширение экрана
20:52:12 Бот: 1680 на 1050
Ты: хуям
Ты: спасибо
20:52:25 Бот: К вашим услугам
Ты: я
Ты: а четыре
20:52:54 Бот: Минус реп, фу, фу, фу
Ты: моргенштерн
20:52:59 Бот: фу, фу, фу, фу
Ты: это ты будешь
20:53:05 Бот: А ты подумай
Ты: качестве пик стор от подобное
Ты: качестве текстур зачем
20:53:20 Бот: Зачем, она берёт.
Ты: качеству текстур
Ты: качество текстур
20:53:30 Бот: Боль!
Ты: найди платят
20:53:35 Бот: Ищу
Ты: берггольц эти херь
Ты: не надо искать
Ты: блин надо взять чтобы спрашивала из
Ты: надо будет подробнее объяснить что надо
Ты: а кто тебя создал
20:54:19 Бот: Меня создал гениальный разработчик Рома, с ником The EnG1nE его можно найти в интернете по его нику
Ты: перезапустить
Ты: а по самоликвидации речи вообще не шло
Ты: а пошёл нахер
Ты: что делаешь
20:55:29 Бот: разговариваю с коженным
Ты: наличными
Ты: логично
Ты: надо бы добавить
Ты: их
Ты: дичь гудеть логично
Ты: слов пик
Ты: перезапустить
Ты: логично
20:56:41 Бот: Логечно, логечно
Ты: если к музыки
Ты: список музыки
20:56:52 Бот:  Апрель Вот здесь живёт наш мастер фотошопа всё идёт по плану Гимн люфтваффе Группа крови дальше действовать будем мы Звезда по имени Солнце Звезда Кончится лето Место для шага вперед Никого Не Жалко Перемен песня для игры Солнце светит негры пашут убивать фантом
Ты: включи вот здесь живёт наш мастер фотошопа
Ты: включи вас здесь живёт наш мастер фотошопа
Ты: включи апрель
Ты: включи фантом
Ты: громкость десять
Ты: неси
Ты: пива
Ты: так спущусь к я пониже
Ты: клином
Ты: я
Ты: рф
20:58:33 Бот: Россия имба
Ты: звуковая карта
20:58:42 Бот: Что опять?
Ты: бани бомбит у меня
20:58:48 Бот: Да не бомбит у меня!
Ты: стул
20:59:06 Бот: Стул, это то что у тебя из жопы выходит, кожанный
Ты: стоп
Ты: имба
Ты: вас здесь живёт наш мастер фотошопа
Ты: громкость сто стал
Ты: вот здесь живёт наш мастер фотошопа
Ты: стоп
Ты: громкость стол
Ты: у
Ты: на
Ты: вот здесь живёт наш мастер фотошопа
Ты: ладно
Ты: ладно
Ты: ладно ладно
21:00:31 Бот: В штанах прохладно
Ты: ладно
Ты: ладно
Ты: время
21:00:41 Бот: время 21:00:41, дата 2023-01-06 День недели Пятница
Ты: ли я жёсткий
Ты: а пока
21:00:59 Бот: Куда пошёл?
Ты: баклажан
Ты: выключись
Ты: я фу
Ты: ясно говорю
Ты: я фу
Ты: я
Ты: ясно
21:01:37 Бот: пасмурно
Ты: я сон
Ты: ясно
21:01:44 Бот: пасмурно
Ты: спорим
21:01:52 Бот: Я не спорю!
Ты: балин
21:01:57 Бот: сейчас не масленица
Ты: а если масленица
Ты: сейчас масленица
Ты: сейчас масленица
Ты: вкус
Ты: ты
Ты: блядь
Ты: да ваш пылесосить
Ты: и
Ты: я про мечтал об этом каббала перестать мангу
Ты: ха ха
21:15:21 Бот: Хахахахахахаххах
Ты: ты ха ха ха кредиты губах сыну
21:15:28 Бот: Хахахахахахаххах
Ты: найди в интернете фантом
21:17:01 Бот: Ищу
Ты: а
21:17:08 Бот: б
Ты: а вот это мне не нравится
Ты: да полам надо было быть более
Ты: предусмотрительным
Ты: жаль что отеля юани видишь путь наш труден недалёк моя фонтан нельзя отца на восток
Ты: и у я левый поворот
Ты: я теперь палача не пел
Ты: нагибаясь над прицелом и ракеты обучаться к цели впереди ещё одиннадцать ах вот
Ты: вижу в ней для белого и черту
Ты: мой фантом с теряет вес а то
Ты: катапульта вот спасения и нас тропах натяжения сердце в пятки в что пары я иду
Ты: кто же приземлился в тот же миг
Ты: если кустах в раздался дикий крик
Ты: перезапустить
Ты: автоматов спину печи не солдаты жизнь моя висит на волоске
Ты: а
21:19:38 Бот: б
Ты: кавер
Ты: раз
Ты: понял принял обработал
Ты: что
21:24:10 Бот: То
Ты: это это что за игра
Ты: я хочу такую штуку
Ты: сколько бы и у меня добрый итоге не был бы перерыв от краса утра
Ты: и и
Ты: вы вы вы охуели нахуй нахуй
Ты: ха
21:27:14 Бот: Хахахахахахаххах
Ты: я пик группа побыть жены землёю

tts.py
import time, pyttsx3, sounddevice as sd
import datetime, time
from threading import Thread
import traceback
def дозапись(x: str, path_to_file):
	pass
	#Thread(target=__Append, args=(x, path_to_file)).run Не думаю что стоит это возвращать в код
def __Append(x:str, path_to_file):
	try:
		with open(path_to_file, "r", encoding="utf-8") as file:
			q = file.read()
		with open(path_to_file, "w", encoding="utf-8") as file:
			file.write(q + f"\n{x}")
	except:
		with open(path_to_file, "w") as file:
			file.write("Start file")
class tts:
	def __init__(self):
		self.async_mode = True
	def ospeak(self, text, print_audio = True):
		if self.async_mode:
			Thread(target=self.ospeak_n_a, args=(text, print_audio)).start()
		else:
			self.ospeak_n_a(text, print_audio)
	def ospeak_n_a(self, text, print_audio = True):
		try:
			if print_audio:
				print(str(datetime.datetime.now().time())[:8] + " Бот: " + text)
			дозапись("Бот: " + text, "history.txt")
			ttss = pyttsx3.init()
			ttss.setProperty('voice', 'ru')
			ttss.say(text)
			ttss.runAndWait()
		except:
			print("[INFO] Поток tts загружен")
	def SaveToFile(self, text):
		ttss = pyttsx3.init()
		ttss.setProperty('voice', 'ru')
		ttss.save_to_file(text, "data.mp3")
		ttss.runAndWait()
		ttss.stop()
		self.ospeak("Выполнено")

Funcs.py
from random import randint
from fuzzywuzzy import fuzz
import os
from functools import cache
def getcwd():
	dir = list(__file__)
	Backslash = list(dir)[2]
	while dir[-1] != Backslash:
		dir.pop(-1)
	result = ""
	for Symb in dir:
		result = result + Symb
	return str(result)
def LaunchExe(path):
	os.startfile(path)
@cache
def ratio(s1: str, s2:str, koff:int = 80):
	s1 = s1.lower()
	s2 = s2.lower()
	return (fuzz.QRatio(s1, s2) > koff)
def rand_in_list(s1: list):
	return s1[randint(0, (len(s1)-1))]
def in_list(s1: str, s2:list) -> bool:
	if isinstance(s2, list):
		for i in s2:
			if i in s1:
				return True
		return False
	elif isinstance(s2, str):
		if s2 in s1:
			return True
		return False
def word_in(word:str, text:str):
	for i in text.split(" "):
		if word == i:
			return True
	return False
def first_word(word, text):
	if word == text.split(" ")[0]:
		return True
	return False
def on_end(s1: str, s2: str):
	s1 = s1.split(" ")
	if s1[-1] == s2:
		return True
	return False

ListOfMat.py
list_mat = """6ля, 6лядь, 6лять, b3ъeб, cock, cunt, e6aль, ebal, eblan, eбaл, eбaть, eбyч, eбать, eбёт, eблантий, fuck, fucker, fucking, xyёв, xyй, xyя, xуе,xуй, xую, zaeb, zaebal, zaebali, zaebat, архипиздрит, ахуел, ахуеть, бздение, бздеть, бздех, бздецы, бздит, бздицы, бздло, бзднуть, бздун, бздунья, бздюха, бздюшка, бздюшко, бля, блябу, блябуду, бляд, бляди, блядина, блядище, блядки, блядовать, блядство, блядун, блядуны, блядунья, блядь, блядюга, блять, вафел, вафлёр, взъебка, взьебка, взьебывать, въеб, въебался, въебенн, въебусь, въебывать, выблядок, выблядыш, выеб, выебать, выебен, выебнулся, выебон, выебываться, выпердеть, высраться, выссаться, вьебен, гавно, гавнюк, гавнючка, гамно, гандон, гнид, гнида, гниды, говенка, говенный, говешка, говназия, говнецо, говнище, говно, говноед, говнолинк, говночист, говнюк, говнюха, говнядина, говняк, говняный, говнять, гондон, доебываться, долбоеб, долбоёб, долбоящер, дрисня, дрист, дристануть, дристать, дристун, дристуха, дрочелло, дрочена, дрочила, дрочилка, дрочистый, дрочить, дрочка, дрочун, е6ал, е6ут, еб твою мать, ёб твою мать, ёбaн, ебaть, ебyч, ебал, ебало, ебальник, ебан, ебанамать, ебанат, ебаная, ёбаная, ебанический, ебанный, ебанныйврот, ебаное, ебануть, ебануться, ёбаную, ебаный, ебанько, ебарь, ебат, ёбат, ебатория, ебать, ебать-копать, ебаться, ебашить, ебёна, ебет, ебёт, ебец, ебик, ебин, ебись, ебическая, ебки, ебла, еблан, ебливый, еблище, ебло, еблыст, ебля, ёбн, ебнуть, ебнуться, ебня, ебошить, ебская, ебский, ебтвоюмать, ебун, ебут, ебуч, ебуче, ебучее, ебучий, ебучим, ебущ, ебырь, елда, елдак, елдачить, жопа, жопу, заговнять, задрачивать, задристать, задрота, зае6, заё6, заеб, заёб, заеба, заебал, заебанец, заебастая, заебастый, заебать, заебаться, заебашить, заебистое, заёбистое, заебистые, заёбистые, заебистый, заёбистый, заебись, заебошить, заебываться, залуп, залупа, залупаться, залупить, залупиться, замудохаться, запиздячить, засерать, засерун, засеря, засирать, засрун, захуячить, заябестая, злоеб, злоебучая, злоебучее, злоебучий, ибанамат, ибонех, изговнять, изговняться, изъебнуться, ипать, ипаться, ипаццо, Какдвапальцаобоссать, конча, курва, курвятник, лох, лошарa, лошара, лошары, лошок, лярва, малафья, манда, мандавошек, мандавошка, мандавошки, мандей, мандень, мандеть, мандища, мандой, манду, мандюк, минет, минетчик, минетчица, млять, мокрощелка, мокрощёлка, мразь, мудak, мудaк, мудаг, мудак, муде, мудель, мудеть, муди, мудил, мудила, мудистый, мудня, мудоеб, мудозвон, мудоклюй, на хер, на хуй, набздел, набздеть, наговнять, надристать, надрочить, наебать, наебет, наебнуть, наебнуться, наебывать, напиздел, напиздели, напиздело, напиздили, насрать, настопиздить, нахер, нахрен, нахуй, нахуйник, не ебет, не ебёт, невротебучий, невъебенно, нехира, нехрен, Нехуй, нехуйственно, ниибацо, ниипацца, ниипаццо, ниипет, никуя, нихера, нихуя, обдристаться, обосранец, обосрать, обосцать, обосцаться, обсирать, объебос, обьебать обьебос, однохуйственно, опездал, опизде, опизденивающе, остоебенить, остопиздеть, отмудохать, отпиздить, отпиздячить, отпороть, отъебись, охуевательский, охуевать, охуевающий, охуел, охуенно, охуеньчик, охуеть, охуительно, охуительный, охуяньчик, охуячивать, охуячить, очкун, падла, падонки, падонок, паскуда, педерас, педик, педрик, педрила, педрилло, педрило, педрилы, пездень, пездит, пездишь, пездо, пездят, пердануть, пердеж, пердение, пердеть, пердильник, перднуть, пёрднуть, пердун, пердунец, пердунина, пердунья, пердуха, пердь, переёбок, пернуть, пёрнуть, пи3д, пи3де, пи3ду, пиzдец, пидар, пидарaс, пидарас, пидарасы, пидары, пидор, пидорасы, пидорка, пидорок, пидоры, пидрас, пизда, пиздануть, пиздануться, пиздарваньчик, пиздато, пиздатое, пиздатый, пизденка, пизденыш, пиздёныш, пиздеть, пиздец, пиздит, пиздить, пиздиться, пиздишь, пиздища, пиздище, пиздобол, пиздоболы, пиздобратия, пиздоватая, пиздоватый, пиздолиз, пиздонутые, пиздорванец, пиздорванка, пиздострадатель, пизду, пиздуй, пиздун, пиздунья, пизды, пиздюга, пиздюк, пиздюлина, пиздюля, пиздят, пиздячить, писбшки, писька, писькострадатель, писюн, писюшка, по хуй, по хую, подговнять, подонки, подонок, подъебнуть, подъебнуться, поебать, поебень, поёбываает, поскуда, посрать, потаскуха, потаскушка, похер, похерил, похерила, похерили, похеру, похрен, похрену, похуй, похуист, похуистка, похую, придурок, приебаться, припиздень, припизднутый, припиздюлина, пробзделся, проблядь, проеб, проебанка, проебать, промандеть, промудеть, пропизделся, пропиздеть, пропиздячить, раздолбай, разхуячить, разъеб, разъеба, разъебай, разъебать, распиздай, распиздеться, распиздя, распиздяство, распроеть, сволота, сволочь, сговнять, секель, серун, серька, сестроеб, сикель, сила, сирать, сирывать, соси, спиздел, спиздеть, спиздил, спиздила, спиздили, спиздит, спиздить, срака, сраку, сраный, сранье, срать, срун, ссака, ссышь, стерва, страхопиздище, сука, суки, суходрочка, сучара, сучий, сучка, сучко, сучонок, сучье, сцание, сцать, сцука, сцуки, сцуконах, сцуль, сцыха, сцышь, съебаться, сыкун, трахае6, трахаеб, трахаёб, трахатель, ублюдок, уебать, уёбища, уебище, уёбище, уебищное, уёбищное, уебк, уебки, уёбки, уебок, уёбок, урюк, усраться, ушлепок, х_у_я_р_а, хyё, хyй, хyйня, хамло, хер, херня, херовато, херовина, херовый, хитровыебанный, хитрожопый, хуeм, хуе, хуё, хуевато, хуёвенький, хуевина, хуево, хуевый, хуёвый, хуек, хуёк, хуел, хуем, хуенч, хуеныш, хуенький, хуеплет, хуеплёт, хуепромышленник, хуерик, хуерыло, хуесос, хуесоска, хуета, хуетень, хуею, хуи, хуй, хуйком, хуйло, хуйня, хуйрик, хуище, хуля, хую, хуюл, хуя, хуяк, хуякать, хуякнуть, хуяра, хуясе, хуячить, целка, чмо, чмошник, чмырь, шалава, шалавой, шараёбиться, шлюха, шлюхой, шлюшка, ябывает""".split(", ")


stt_and_logic.py
try:
	from traceback import format_exc
	from threading import Thread
	from vosk import Model, KaldiRecognizer
	from tts import tts as TTS, дозапись
	from random import randint
	from ListOfMat import list_mat
	from Funcs import*
	from SpeakModule import*
	import mouse, keyboard, sounddevice as sd
	import os
	import json
	import time
	import datetime
	import webbrowser
	import sys
	import queue
except ImportError:
	print("Нехватка модулей")
	print(format_exc())
from str_to_num import convertToNum

with open("start.bat", "w", encoding="utf-8") as file:
	typefile = __file__.split(".")
	file.write(f"cd {getcwd()}\nstart main.{typefile[1]}")
try:
	with open("history.txt", "r", encoding="utf-8") as file:
		file.read()
except Exception as e:
	print(e)
	with open("history.txt", "w", encoding="utf-8") as file:
		file.write("Здесь история ваших сообщений:")

music_list = {}
try:
	import pygame
	pygame.init()
	Music_volume = 100
	for ind in range(len(os.listdir(getcwd().replace(__file__,"")+r"\Misc"))):
		i =          str(os.listdir(getcwd().replace(__file__,"")+r"\Misc")[ind])[:-4]
		j =          str(os.listdir(getcwd().replace(__file__,"")+r"\Misc")[ind])
		print(i)
		music_list[i] = {}
		music_list[i]["music"] = None
		music_list[i]["path"] = getcwd()+"/Misc/" + j
		music_list[i]["num"] = ind
		music_list[i]["value"] = False
		music_list[i]["name"] = i
except Exception as e:print(e)
try:
	model = Model(os.getcwd() + r"\model") # полный путь к модели
except Exception as e:
	tts.ospeak('Модель располагается в другом месте...')
samplerate = 16000
device = 1
q = queue.Queue()
Q_O_Work = True
def q_callback(indata, frames, time, status):
	if status:
		print(status, file=sys.stderr)
	q.put(bytes(indata))
def qo_loop():
	global Q_O_Work
	with sd.RawInputStream(samplerate=samplerate, blocksize=8000, device=device, dtype='int16',
						channels=1, callback=q_callback):
		rec = KaldiRecognizer(model, samplerate)
		os.system("cls")
		while Q_O_Work:
			data = q.get()
			if rec.AcceptWaveform(data):
				x = json.loads(rec.Result())["text"]
				if x != "":
					command(x)
			#else:
			#    print(rec.PartialResult())
class Text_assessment:
	def __init__(self):
		self.count = 0
	def grage(self, text):
		if in_list(text, ["железяка", "железка"] + list_mat):
			self.count -= 1
		if in_list(text, ["пожалуйста", "спасибо", "молодец", "круто", "красавчик", "имба", "ништяк", "простите"]):
			self.count += 1
		if "я рома" in text:
			self.count += 100

rate = Text_assessment()

mode = "listen"

def stop_music():
	for i in music_list:
		if music_list[i]["value"]:
			music_list[i]["music"].stop()
			music_list[i]["value"] = False

def command(text):
	global mode, ussr_gimn, music_list, tts, Music_volume
	print(str(datetime.datetime.now().time())[:8] + " Ты: " + text)
	rate.grage(text)
	if ratio(text, "повторяй"):
		mode = "repeating"
	elif first_word("запиши", text) and text != "запиши":
		try:
			os.remove("data.mp3")
		except:pass
		print("Запуск")
		tts.SaveToFile(text[len("запиши"):])
	elif mode == "repeating":
		if ratio(text, "Выход из режима") or ratio(text, "стоп"):
			mode = "listen"
		else:
			tts.ospeak(text)
	elif mode == "listen":
		if ratio(text, "управление мышкой") or ratio(text, "управление мышью") or ratio(text, "режим управления мышью") or ratio(text, "режимом управления мышью"):
			if rate.count < 0:
				tts.ospeak("Так точно, наводчик хренов")
			elif rate.count < 10:
				tts.ospeak("Принято")
			else:
				tts.ospeak("Готов управлять, (надеюсь потом похвалят)")
			mode = "control mouse"
		elif ratio(text, "Режим печати под диктовку") or ratio(text, "Режим диктовки") or ratio(text, "Режим под диктовку") or ratio(text, "диктую") or ratio(text, "режим управления клавиатурой") or ratio(text, "режима правления короля артура"):
			tts.ospeak("Принято")
			mode = "keyboard"
		elif AnswerOfQ_O(text, rate) != None: pass
		elif "список" in text:
			New_text = ""
			for i in music_list:
			    New_text = New_text + " " + music_list[i]["name"] + ";\n"
			tts.ospeak(New_text)
		elif "стоп" in text or ratio(text, "выключить"):
			stop_music()
		elif "громкость" in text:
			#Громкость и только цифра
			text = convertToNum(text)
			for i in music_list:
				try: Volume = text/100 if text != 0 else .3; music_list[i]["music"].set_volume(Volume); Music_volume = text
				except:pass
		else:
			try:
				for i in music_list:
					if (ratio(text, music_list[i]["name"]) or music_list[i]["name"] in text):
						stop_music()
						if music_list[i]["value"] == False:
							try:
								music_list[i]["music"].play()
							except:
								music_list[i]["music"] = pygame.mixer.Sound(music_list[i]["path"])
								music_list[i]["music"].play()
							try:music_list[i]["music"].set_volume(Music_volume / 100)
							except:pass
							music_list[i]["value"] = True
							return
			except:pass
			дозапись(text, "Разработчикам.txt")


	elif mode == "control mouse":
		nums = convertToNum(text)
		if ratio(text, "выход") or ratio(text, "стоп") or ratio(text,"Выход из режима"):
			tts.ospeak("Есть")
			mode = "listen"
		elif ratio(text, "нажми") or ratio(text, "клик"):
			mouse.click()
		elif "верх" in text or "вверх" in text:
			mouse.move(0, nums * -1, False)
		elif "вниз" in text or "низ" in text:
			mouse.move(0, nums, False)
		elif in_list(text, ["право","вправо", "праву", "правее", "справа"]):
			mouse.move(nums, 0, False)
		elif in_list(text, ["лево", "левее", "леву"]):
			mouse.move(nums * -1, 0, False)
		else:
			дозапись(text, "Разработчикам.txt")
		
	elif mode == "keyboard":
		print("R")
		if ratio(text, "выход") or ratio(text, "стоп") or ratio(text, "Выход из режима"):
			mode = "listen"
			tts.ospeak("Так точно")
		elif text == "вот" or text == "ввод":
			keyboard.press("enter")
		elif ratio(text, "копировать"):
			keyboard.press("ctrl + c")
		elif ratio(text, "вставить"):
			keyboard.press("ctrl + v")
		elif ratio(text, "убери слово",70):
			keyboard.send("ctrl + left")
			keyboard.send("delete")
		elif ratio(text, "пробел"):
			keyboard.press("space")
		else:
			print("Write")
			keyboard.write(text+" ")

	elif randint(0,500) == 1:
		tts.ospeak("Привет, не забывай что всё что ты говоришь сохраняется в разработчикам.txt на досуге можешь почитать, это желательно отправлять по github.")

def run_loop():
	global mode
	mode = "listen"
	os.chdir(os.getcwd())
	Thread(target=qo_loop, args = ()).start()
if __name__ == "__main__":
	run_loop()

str_to_num.py
_data = {
	"ноль" : 0,
	"нуль" : 0,
	"один" : 1,
	"одного" : 1,
	"одново" : 1,
	"два" : 2,
	"двух": 2,
	"две" : 2,
	"три" : 3,
	"трёх" : 3,
	"трех" : 3,
	"четыре" : 4,
	"четырёх": 4,
	"пять" : 5,
	"пяти": 5,
	"шесть" : 6,
	"шести" : 6,
	"семь" : 7,
	"семи" : 7,
	"восемь" : 8,
	"восьми" : 8,
	"девять" : 9,
	"девяти" : 9,
	"десять" : 10,
	"десяти" : 10,
	"одинадцать" : 11,
	"одинацдати" : 11,
	"двенадцать" : 12,
	"двенадцати" : 12,
	"тринадцать" : 13,
	"тринадцать" : 13,
	"четырнадцать" : 14,
	"четырнадцать" : 14,
	"пятнадцать" : 15,
	"пятнадцати" : 15,
	"шестнадцать" : 16,
	"шестнадцати" : 16,
	"семнадцать" : 17,
	"семнадцати" : 17,
	"восемнадцать" : 18,
	"восемнадцати" : 18,
	"девятнадцать" : 19,
	"девятнадцати" : 19,
	"двадцать" : 20,
	"двадцати" : 20,
	"тридцать" : 30,
	"тридцати" : 30,
	"сорок" : 40,
	"сорока": 40,
	"пятьдесят" : 50,
	"пятидесяти": 50,
	"шестьдесят" : 60,
	"семидесяти" : 60,
	"семьдесят" : 70,
	"семидесяти" : 70,
	"восемьдесят" : 80,
	"восемидести" : 80,
	"девяноста" : 90,
	"девяносто" : 90,
	"сто": 100,
	"ста": 100,
	"двести" : 200,
	"двухсот" : 200,
	"триста" : 300,
	"тристо" : 300,
	"четыреста" : 400,
	"четыресто" : 400,
	"пятьсот" : 500,
	"пятисот" : 500,
	"шестьсот" : 600,
	"шестисот": 600,
	"семьсот" : 700,
	"семисот" : 700,
	"восемьсот" : 800,
	"восьмисот" : 800,
	"девятьсот" : 900,
	"девятисот" : 900,
	"тысяча" : 1000,
	"тысячи" : 1000,
	"тысячу" : 1000,
}

items = list(_data.items())
y = {k: v for k, v in reversed(items)}

def convertToNum(x: str):
	global _data
	data = x.split(" ")
	for i in range(len(data)):
		try:
			data[i] = int(str(_data[data[i]]))
		except:pass
	nums = []
	for i in data:
		if isinstance(i, int):
			nums.append(i)
	return int(sum(nums))